### PR TITLE
Expose lu_factor_batched_cublas

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73878
* __->__ #73877
* #74046

We had bindings for this function, but they were not exposed as a
function inside ATen. This PR exposes them.

This function is tested in the next PR of the stack, where it is used in
`linalg.lu_factor`.

### CODE REVIEW

**Comment by IvanYashchuk** [comment]
`getrfBatched` is used in `linalg.inv`: https://github.com/pytorch/pytorch/blob/bd13bc667e9a56e95b14bb3da88f170548880b91/aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp#L293-L295

Also the description of the PR that added this function says `getrfBatched` doesn't have a good performance: https://github.com/pytorch/pytorch/pull/42403.
Let's see what your performance comparison would reveal.

**Comment by Lezcano** [comment]
For what is worth, that PR just tested up to batches of 4 matrices, which is not very representative either.

File: aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp [hunk 1]
```diff
@@ -91,6 +91,33 @@ void geqrf_batched_cublas(const Tensor& input, const Tensor& tau) {
...
+  auto n = cuda_int_cast(A.size(-2), "n");
+  auto lda = cuda_int_cast(std::max<int>(1, n), "lda");
+
+  auto pivots_data = get_pivots ? pivots.data_ptr<int>() : nullptr;
+  auto infos_data = infos.data_ptr<int>();
+  Tensor a_ptr_array = get_device_pointers<scalar_t>(A);
+  auto a_ptr_array_data = reinterpret_cast<scalar_t**>(a_ptr_array.data_ptr());
+
+  at::cuda::blas::getrfBatched(n, a_ptr_array_data, lda, pivots_data, infos_data, batch_size);
+#endif
```
Comment (nikitaved) [inline aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp:112]: This one handles non-contiguous inputs, right?

Comment (Lezcano) [reply aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp:112]: non-contiguous inputs are handled in the parent fucntion, that is, in `linalg_lu_factor`

**Review by nikitaved — APPROVED** [review]
LGTM! Thanks!

**Review by IvanYashchuk — APPROVED** [review]

**Review by mruberry — APPROVED** [review]
Stamped!

**Pull request closed without merging**